### PR TITLE
Introduce libsks for PKCS#11

### PR DESCRIPTION
**Pull request by etienne-lms**
Introduce libsks: basic source/header/make files and library API: **pkcs11.h**.
All API functions are stubbed.

This P-R introduces the first commits from https://github.com/OP-TEE/optee_client/pull/138.

### CODE REVIEW

File: libsks/Makefile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+LIB_MAJOR		:= $(LIB_NAME).so.$(MAJOR_VERSION)
+LIB_MAJ_MIN		:= $(LIB_NAME).so.$(MAJOR_VERSION).$(MINOR_VERSION)
+LIBSKS_SO_LIBRARY	:= $(LIB_MAJ_MIN)
+LIBSKS_AR_LIBRARY	:= $(LIB_NAME).a
+
+LIBSKS_SRC_DIR		:= src
+
+LIBSKS_SRCS		= pkcs11_api.c
+
+LIBSKS_INCLUDES		= ${CURDIR}/include
```
Comment (pokitoz) [inline libsks/Makefile]: indentation issue?

Comment (etienne-lms) [reply libsks/Makefile]: the fuzzy indentation is due to github interface, not to the patch itself. The file uses 8-char tabulation based indentation.

**Comment by jenswi-linaro** [comment]
I think we need to use the `SKS_CMD_PING` command in some way.

**Comment by etienne-lms** [comment]
I propose to go straight on reviewing https://github.com/OP-TEE/optee_client/pull/138 which uses the `PING` command.

**Comment by etienne-lms** [comment]
superseded by https://github.com/OP-TEE/optee_client/pull/138

**Pull request closed without merging**